addons_tests/test_module/models/hop_dong.py

```diff
@@ -1,19 +1,74 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from datetime import *
+from odoo.exceptions import ValidationError
+
+
+def get_signed_date():
+    signed_date = datetime.now() + timedelta(weeks=1)
+    if signed_date.weekday() == 5:
+        signed_date += timedelta(days=2)
+    elif signed_date.weekday() == 6:
+        signed_date += timedelta(days=1)
+    return signed_date
 
 
 class hop_dong(models.Model):
     _name = 'hop.dong'
     _description = 'hop dong'
 
-    name = fields.Char(string='ten')
-    start_date = fields.Datetime(string='ngay bat dau')
-    end_date = fields.Datetime(string='ngay ket thuc')
-    contract_type = fields.Char(string='loai hop dong')
-    signed_date = fields.Datetime(string='ngay ky')
-    salary_rack = fields.Float(string='luong ngach bac')
-    efficiency_wage = fields.Float(string='luong hieu qua')
+    name = fields.Char(string='Name')
+    start_date = fields.Datetime(string='Start Date: ', default=fields.Datetime.now, format='%Y-%m-%d')
+    end_date = fields.Datetime(string='End Date: ')
+    contract_type = fields.Char(string='Contract Type: ')
+    signed_date = fields.Datetime(string='Signed Date: ', default=get_signed_date())
+    salary_rack = fields.Float(string='Standard Salary: ')
+    efficiency_wage = fields.Float(string='Efficiency Wage: ')
     status = fields.Selection([
-        ('draft', 'ban nhap'),
-        ('ongoing', 'dang dien ra'),
-        ('expired', 'het han')],
-        string='trang thai', default='draft')
+        ('new', 'New'),
+        ('running', 'Running'),
+        ('expired', 'Expired'),
+        ('pause', 'Pause')],
+        string='Status', default='new')
+
+    employee_id = fields.Many2one('hr.employee', string='Employee')
+
+    employee_name = fields.Char(string='Employee Name', required=True, compute='_compute_employee_name')
```

Comment:
employee_name  có lưu trong database không em nhỉ? khi nào thì employee_name có giá trị?

Reply:
employee_name không được lưu trong DB vì nó là trường compute field và nó có giá trị khi employee_id có tồn tại trường name ạ

Reply:
thế khi nào thì employee_name được tính toán lại

Reply:
Biến này sẽ được tính toán lại khi các trường liên quan đến nó có sự thay đổi hoặc có 1 bản ghi mới được tạo ra ạ

Reply:
Ngoài bản ghi mới tạo ra thì còn sự kiện nào gọi hàm này không?

Reply:
khi record.employee_id.name thay đổi giá trị
khi gọi compute() hoặc sudo().compute() thì sẽ tính toán lại ạ